ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/AddGraphicsWithRenderer/AddGraphicsWithRenderer.cpp

```diff
@@ -0,0 +1,190 @@
+
+  createGraphicOverlayWithGraphicAndSymbol(polygonGraphic, polygonFillSymbol);
+}
+
+void AddGraphicsWithRenderer::addCurveGraphic()
+{
+  Geometry heartShapedCurve = createHeart();
+
+  Graphic* curveGraphic = new Graphic(heartShapedCurve);
+
+  SimpleLineSymbol* curveLineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor("black"), 1, this);
+  SimpleFillSymbol* curveFillSymbol = new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor("red"), curveLineSymbol, this);
+
+  createGraphicOverlayWithGraphicAndSymbol(curveGraphic, curveFillSymbol);
+}
+
+void AddGraphicsWithRenderer::createGraphicOverlayWithGraphicAndSymbol(Graphic* graphic, Symbol* symbol)
+{
+  GraphicsOverlay* graphicOverlay = new GraphicsOverlay(this);
+  // set the renderer of the overlay to be the marker symbol
+  graphicOverlay->setRenderer(new SimpleRenderer(symbol, this));
+  // add the graphic to the overlay
+  graphicOverlay->graphics()->append(graphic);
+  // add the overlay to the mapview
+  m_mapView->graphicsOverlays()->append(graphicOverlay);
+}
+
+
+Geometry AddGraphicsWithRenderer::createHeart()
+{
+  // Declare common variables
+  Point origin(40e5, 5e5, 0, SpatialReference::webMercator());
+  SpatialReference spatialReference = origin.spatialReference();
+  int sideLength = 10e5;
+  int minX = origin.x() - 0.5 * sideLength;
+  int minY = origin.y() - 0.5 * sideLength;
+  int arcRadius = sideLength * 0.25;
+
+  // The heart-shape comprises 4-curve segments: 2 bezier curves make the base and 2 arc curves make the top
+  // Declare all points used to create heart-shaped curve, starting at bottom tip and moving clockwise
+  Point bottomTip(origin.x(), minY, spatialReference);
+  Point leftControlPoint1(origin.x(), minY + 0.25 * sideLength, spatialReference);
+  Point leftControlPoint2(minX, origin.y(), spatialReference);
+  Point leftCurveEnd(minX, minY + 0.75 * sideLength, spatialReference);
+  Point leftArcCentre(minX + 0.25 * sideLength, minY + 0.75 * sideLength, spatialReference);
+  Point arcIntersection(origin.x(), minY + 0.75 * sideLength);
+  Point rightArcCentre(minX + sideLength, minY + 0.75 * sideLength, spatialReference);
+  Point rightCurveStart(minX + sideLength, minY + 0.75 * sideLength, spatialReference);
+  Point rightControlPoint1(minX + sideLength, origin.y(), spatialReference);
+  Point rightControlPoint2 = leftControlPoint1;
+
+  // Create curve segments
+  CubicBezierSegment leftCurve(bottomTip, leftControlPoint1, leftControlPoint2, leftCurveEnd, spatialReference);
+  CubicBezierSegment rightCurve(rightCurveStart, rightControlPoint1, rightControlPoint2, bottomTip, spatialReference);
+  EllipticArcSegment leftArc(leftCurveEnd, arcIntersection, 0, false, false, arcRadius, 1, spatialReference);
+  EllipticArcSegment rightArc(arcIntersection, rightCurveStart, 0, false, false, arcRadius, 1, spatialReference);
+
+  // Create part from segments
+  Part* heart = new Part(spatialReference, this);
```

Comment:
Generally our geometry types are not handled as pointer types (e.g. derived from `QObject`) - that might not be the case with curves but just good to double-check